### PR TITLE
Support Speculation Rules API formatting in HTML

### PR DESCRIPTION
**Pull request by sosukesuzuki**
## Description

Support Speculation Rules API in HTML.

**refs:**

- https://github.com/WICG/nav-speculation
- https://web.dev/speculative-prerendering/

## Checklist

- [x] I’ve added tests to confirm my change works.
- [ ] (If the change is user-facing) I’ve added my changes to `changelog_unreleased/*/XXXX.md` file following `changelog_unreleased/TEMPLATE.md`.
- [x] I’ve read the [contributing guidelines](https://github.com/prettier/prettier/blob/main/CONTRIBUTING.md).

**✨[Try the playground for this PR](https://prettier.io/playground-redirect)✨**

### CODE REVIEW

File: src/language-html/utils/index.js [hunk 1]
```diff
@@ -364,7 +364,10 @@ function _inferScriptParser(node) {
     return "html";
   }
 
-  if (type && (type.endsWith("json") || type.endsWith("importmap"))) {
+  if (
+    (type && (type.endsWith("json") || type.endsWith("importmap"))) ||
```
Comment (fisker) [inline src/language-html/utils/index.js:368]: Not related, but is `importmap` check a mistake? I think it should be  `=== "importmap"`

Comment (fisker) [reply src/language-html/utils/index.js:368]: Maybe this the `json` mime check should be endsWith `/json` or `+json`?

**Review by fisker — APPROVED** [review]

File: src/language-html/utils/index.js (hunk 1 above)
Comment (sosukesuzuki) [reply src/language-html/utils/index.js:368]: > Not related, but is importmap check a mistake? I think it should be === "importmap"

Maybe for `systemjs-importmap`. https://github.com/systemjs/systemjs/blob/main/docs/import-maps.md#loading-import-maps

**Review by lipis — APPROVED** [review]

**Review by Deloww — COMMENTED** [review]
Update

**Review by Deloww — COMMENTED** [review]
changelog_unreleased/html/12882.md

**Pull request merged**